src/magic_uv/op/texture_wrap.py

```diff
@@ -238,26 +227,65 @@ def execute(self, context):
                 self.report({'WARNING'}, "More than 1 vertex must be unshared")
                 return {'CANCELLED'}
 
-            # get target info
+            tform_mtx = None
+            for other_vert in ref_other_verts:
+                # get reference info
+                a_3d = common_verts[0]["vert"].co
+                b_3d = common_verts[1]["vert"].co
+                c_3d = other_vert["vert"].co
+                a_uv = common_verts[0]["ref_loop"][uv_layer].uv
+                b_uv = common_verts[1]["ref_loop"][uv_layer].uv
+                c_uv = other_vert["loop"][uv_layer].uv
+
+                # AB = shared edge, C = third vert of ref face
+                # set up a 2D coordinate system with coordinates relative to AB
+                # X = C projected onto AB, XC/AX = perpendicular/parallel to AB
+                xc_3d, x_3d = common.diff_point_to_segment(a_3d, b_3d, c_3d)
+                ax_3d = x_3d - a_3d
+                ab_3d = b_3d - a_3d
+                ab_2d = Vector((0.0, ab_3d.length))
+                ac_2d = Vector((xc_3d.length,
+                                math.copysign(ax_3d.length, ax_3d.dot(ab_3d))))
+                ab_uv = b_uv - a_uv
+                ac_uv = c_uv - a_uv
+
+                # extra check for collinear verts
+                if xc_3d.length < 1e-5:
+                    continue
```

Comment:
Is it possible to change the hard coded value to the constant variable for changing this value easily in future?